plugin/pkg/shoot/validator/admission.go

```diff
@@ -364,32 +364,53 @@ func (c *validationContext) validateSeedSelectionForHAShoot() error {
 	return nil
 }
 
-func (c *validationContext) validateScheduling(ctx context.Context, a admission.Attributes, authorizer authorizer.Authorizer, shootLister corelisters.ShootLister) error {
-	if a.GetOperation() == admission.Delete {
-		return nil
-	}
-
-	// If there is already a seedName in the shoot yaml, the scheduler will not create the Binding
-	// So we need to keep the validation here.
+func (c *validationContext) validateScheduling(ctx context.Context, a admission.Attributes, authorizer authorizer.Authorizer, shootLister corelisters.ShootLister, seedLister corelisters.SeedLister, shootStateLister corev1alpha1listers.ShootStateLister) error {
 	var (
 		shootIsBeingScheduled          = c.oldShoot.Spec.SeedName == nil && c.shoot.Spec.SeedName != nil
 		shootIsBeingRescheduled        = c.oldShoot.Spec.SeedName != nil && c.shoot.Spec.SeedName != nil && *c.shoot.Spec.SeedName != *c.oldShoot.Spec.SeedName
 		mustCheckSchedulingConstraints = shootIsBeingScheduled || shootIsBeingRescheduled
 	)
 
-	if shootIsBeingScheduled {
-		if a.GetOperation() == admission.Create {
+	switch a.GetOperation() {
+	case admission.Create:
+		if shootIsBeingScheduled {
 			if err := authorize(ctx, a, authorizer, "set .spec.seedName"); err != nil {
 				return err
 			}
 		}
+	case admission.Update:
+		if a.GetSubresource() == "binding" {
+			if c.oldShoot.Spec.SeedName != nil && c.shoot.Spec.SeedName == nil {
+				return admission.NewForbidden(a, fmt.Errorf("spec.seedName cannot be set to nil"))
+			}
+
+			if shootIsBeingRescheduled {
+				if !utilfeature.DefaultFeatureGate.Enabled(features.SeedChange) {
+					if err := apivalidation.ValidateImmutableField(c.oldShoot.Spec.SeedName, c.shoot.Spec.SeedName, field.NewPath("spec", "seedName")).ToAggregate(); err != nil {
+						return err
+					}
+				}
+
+				newShootSpec := c.shoot.Spec
+				newShootSpec.SeedName = c.oldShoot.Spec.SeedName
+				if !reflect.DeepEqual(newShootSpec, c.oldShoot.Spec) {
+					return admission.NewForbidden(a, fmt.Errorf("no Shoot spec other than .spec.seedName should be changed with the binding subresource"))
```

Comment:
Nit to make the error text a bit clearer for when the spec change is forbidden:
```suggestion
					return admission.NewForbidden(a, fmt.Errorf("only spec.seedName can be changed when using the binding subresource and the shoot is being rescheduled to a new seed"))
```

Reply:
how about "only spec.seedName can be changed using the binding subresource when the shoot is being rescheduled to a new seed"?

Reply:
Yeah, works too :).